modules/analysis/aws-batch-demo/stack.py

```diff
@@ -0,0 +1,175 @@
+            removal_policy=RemovalPolicy.DESTROY,
+            point_in_time_recovery=True,
+            stream=dynamo.StreamViewType.NEW_AND_OLD_IMAGES,
+        )
+
+        # Create Dag IAM Role and policy
+        policy_statements = [
+            iam.PolicyStatement(
+                actions=["dynamodb:*"],
+                effect=iam.Effect.ALLOW,
+                resources=[f"arn:aws:dynamodb:{self.region}:{self.account}:table/{dep_mod}*"],
+            ),
+            iam.PolicyStatement(
+                actions=["ecr:*"],
+                effect=iam.Effect.ALLOW,
+                resources=[f"arn:aws:ecr:{self.region}:{self.account}:repository/{dep_mod}*"],
+            ),
+            iam.PolicyStatement(
+                actions=["s3:*"],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
+            ),
+            iam.PolicyStatement(
+                actions=[
+                    "batch:UntagResource",
+                    "batch:DeregisterJobDefinition",
+                    "batch:TerminateJob",
+                    "batch:CancelJob",
+                    "batch:SubmitJob",
+                    "batch:RegisterJobDefinition",
+                    "batch:TagResource",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=[
+                    f"arn:aws:batch:{self.region}:{self.account}:job-queue/addf*",
+                    f"arn:aws:batch:{self.region}:{self.account}:job-definition/*",
+                    f"arn:aws:batch:{self.region}:{self.account}:job/*",
+                ],
+            ),
+            iam.PolicyStatement(
+                actions=[
+                    "iam:PassRole",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=[
+                    f"arn:aws:iam::{self.account}:role/addf*",
+                ],
+            ),
+            iam.PolicyStatement(
+                actions=[
+                    "batch:Describe*",
+                    "batch:List*",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=[
+                    "*",
+                ],
+            ),
+        ]
+        dag_document = iam.PolicyDocument(statements=policy_statements)
+
+        batch_role_name = f"{dep_mod}-dag-role"
+
+        self.dag_role = iam.Role(
+            self,
+            f"dag-role-{dep_mod}",
+            assumed_by=iam.CompositePrincipal(
+                iam.ArnPrincipal(self.mwaa_exec_role),
+                iam.ServicePrincipal("ecs-tasks.amazonaws.com"),
+            ),
+            inline_policies={"DagPolicyDocument": dag_document},
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AmazonECSTaskExecutionRolePolicy"),
+                iam.ManagedPolicy.from_managed_policy_arn(
+                    self, id="fullaccess", managed_policy_arn=self.full_access_policy
+                ),
+            ],
```

Comment:
This covers read/write access to all of the datalake buckets, I think for a demo this is ok as the airflow dag could have additional steps that move data from raw to intermediate and then to curated. But yeah I agree this could be more fine-grained, then I would need to refactor the datalake-buckets module to export the KMS arns or individual read/write policies for all of the buckets.